patched-vscode/extensions/sagemaker-extension/src/extension.ts

```diff
@@ -71,19 +71,21 @@ function initialize(sagemakerCookie: SagemakerCookie) {
     const currentTime = Date.now();
     const timeToExpiry = getExpiryTime(sagemakerCookie) - currentTime;
 
-    if (timeToExpiry <= 0) {
-        signInError(sagemakerCookie);
-    } else if (timeToExpiry >= FIFTEEN_MINUTES_INTERVAL_MILLIS) {
-        const warningTime = timeToExpiry - FIFTEEN_MINUTES_INTERVAL_MILLIS;
-        setTimeout(() => {
-            showWarningDialog();
-        }, warningTime);
-    } else {
-        // If less than or equal to 15 minutes left, set a timer for the remaining time
-        const warningTime = timeToExpiry % FIVE_MINUTES_INTERVAL_MILLIS;
-        setTimeout(() => {
-            showWarningDialog();
-        }, warningTime);
+    if (getExpiryTime(sagemakerCookie) !== -1) {
+        if (timeToExpiry <= 0) {
```

Comment:
curious, why do you use getExpiryTime() in 74 and timeToExpiry in 75, 77, 84, etc? is there a difference?

Reply:
`getExpiryTime` and `timeToExpiry` refer to different values
- `getExpiryTime` - is the value of the local `expiryTime` set by Code Editor. When the cookie is present, it gives the UNIX time of the expiry time (8 hours ahead of app start up). When no cookie is present, it returns `-1`
- `timeToExpiry` - is the UNIX expiry time - the current time (8 hours). We use this to compare against 15 min to provide the warning